src/compiler/emitter.ts

```diff
@@ -365,8 +365,10 @@ namespace ts {
             }
             const version = ts.version; // Extracted into a const so the form is stable between namespace and module
             const buildInfo: BuildInfo = { bundle, program, version };
+            const buildInfoText = getBuildInfoText(buildInfo);
+            host.buildInfoCallbacks?.onWrite(buildInfoText.length);

```

Comment:
Technically the length is going to report the number of 16-byte character codes, right? But the build info is written in utf-8. There's a theoretical discrepancy here. Maybe every place we record file length has a similar problem.

Reply:
> If so, I think it might actually be preferable to report the character count, rather than the byte count

Yeah, but `length` doesn't report the number of characters - it reports the number of code units, which is sometimes different.

Regardless, I think your point about being consistent everywhere is fine.

Reply:
I think having some consistent measure is good enough. I would like to get this in so that i can use this without having to merge into my branches temporarily.